src/k8s/pkg/k8sd/setup/certificates.go

```diff
@@ -11,54 +11,73 @@ import (
 )
 
 // ensureFile creates fname with the specified contents, mode and owner bits.
-// ensureFile will delete the file if contents is an empty string.
-func ensureFile(fname string, contents string, uid, gid int, mode fs.FileMode) error {
+// It will delete the file if contents is an empty string.
+// It returns true if a file was updated (or deleted) and any error that occured.
+func ensureFile(fname string, contents string, uid, gid int, mode fs.FileMode) (bool, error) {
 	if contents == "" {
-		if err := os.Remove(fname); err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("failed to delete: %w", err)
+		if err := os.Remove(fname); err != nil {
+			return false, fmt.Errorf("failed to delete: %w", err)
 		}
-		return nil
+
+		return true, nil
```

Comment:
this does not look right to me. `ensureFile(path, "", ...)` will now fail if the file does not exist. but: we do not want to return an error if the file does not exist, it is exactly what we asked `ensureFile` to do.

consider something like this instead. see how we handle the file does not exist case:

```go
if err := os.Remove(fname); err != nil {
  if !os.IsNotExist(err) {
    return false, fmt.Errorf("failed to delete: %w", err)
  }
  return false, nil
}

return true, nil
```